static/app/components/commandPalette/types.tsx

```diff
@@ -0,0 +1,28 @@
+import type {ReactNode} from 'react';
+import type {LocationDescriptor} from 'history';
+
+export type CommandPaletteAction = {
```

Comment:
I think there are a couple of things that are mixed here that make the definitions a bit unreadable.

How about something like
```
{ 
key, group,keywords (+other action meta as top level keys)
and action a separate key that controls the rendering pieces
}
```

I would have honestly also preferred if we could delegate rendering to every place that registers the action